matrix-appservice-irc/config.yaml

```diff
@@ -0,0 +1,508 @@
+        # whitelist:
+        #   - "@foo:example.com"
+        #   - "@bar:example.com"
+        #
+        # Prevent the given list of channels from being mapped under any
+        # circumstances.
+        # exclude: ["#foo", "#bar"]
+
+      # excludedUsers:
+      #   - regex: "@.*:evilcorp.com"
+      #     kickReason: "We don't like Evilcorp"
+
+      # Configuration for controlling how Matrix and IRC membership lists are
+      # synced.
+      membershipLists:
+        # Enable the syncing of membership lists between IRC and Matrix. This
+        # can have a significant effect on performance on startup as the lists are
+        # synced. This must be enabled for anything else in this section to take
+        # effect. Default: false.
+        enabled: false
+
+        # Syncing membership lists at startup can result in hundreds of members to
+        # process all at once. This timer drip feeds membership entries at the
+        # specified rate. Default: 10000. (10s)
+        floodDelayMs: 10000
+
+        global:
+          ircToMatrix:
+            # Get a snapshot of all real IRC users on a channel (via NAMES) and
+            # join their virtual matrix clients to the room.
+            initial: false
+            # Make virtual matrix clients join and leave rooms as their real IRC
+            # counterparts join/part channels. Default: false.
+            incremental: false
+
+          matrixToIrc:
+            # Get a snapshot of all real Matrix users in the room and join all of
+            # them to the mapped IRC channel on startup. Default: false.
+            initial: false
+            # Make virtual IRC clients join and leave channels as their real Matrix
+            # counterparts join/leave rooms. Make sure your 'maxClients' value is
+            # high enough! Default: false.
+            incremental: false
+
+        # Apply specific rules to Matrix rooms. Only matrix-to-IRC takes effect.
+        rooms:
+          - room: "!fuasirouddJoxtwfge:localhost"
+            matrixToIrc:
+              initial: false
+              incremental: false
+
+        # Apply specific rules to IRC channels. Only IRC-to-matrix takes effect.
+        channels:
+          - channel: "#foo"
+            ircToMatrix:
+              initial: false
+              incremental: false
+
+      mappings:
+        # 1:many mappings from IRC channels to room IDs on this IRC server.
+        # The matrix room must already exist. Your matrix client should expose
+        # the room ID in a "settings" page for the room.
+        "#test-matrix":
+          roomIds: ["!YAkpfBukqCvewVzbVb:matrix.ocf.berkeley.edu"]
+          # Channel key/password to use. Optional. If provided, matrix users do
+          # not need to know the channel key in order to join the channel.
+          # key: "secret"
+        "#matrix":
+          roomIds: ["!zEhTjofIhaNlSpzxlN:matrix.ocf.berkeley.edu"]
```

Comment:
shouldn't need this if we're automatically mapping, right?

Reply:
I would imagine the process of joining channels to be confusing for first time Matrix joiners, who might be inclined to check the room list for channels. If they're not public, they might be confused on what to do in a seemingly empty server. I'm going to have to play around with the user experience.

One of my ideas was that we could try expanding this to default starter channels (so #announcements, #ocf, #rebuild, etc.) and that other channels that weren't manually mapped would be joined using the bridge.

Reply:
i think you should be able to add autobridge channels to the public directory in matrix